### PR TITLE
Improve elevation layers handling with StaticProvider

### PR DESCRIPTION
**Pull request by peppsac**
## Description
2 commits fixing elevation layer processing.

The first one reorganize the code a bit to avoid needless work.
The second fix StaticProvider to use the proper download function.

### CODE REVIEW

**Comment by zarov** [comment]
With the `mimetype` doomed to disappear, it may be a better option to use `layer.format` instead, no ?
Thanks for this cleaning anyway !

**Comment by peppsac** [comment]
Can someone review this short PR please? Thanks!

File: src/Core/Scheduler/Providers/StaticProvider.js [hunk 1]
```diff
@@ -95,17 +102,29 @@ export default {
                     extent,
                 });
             }
+        }).then(() => {
+            if (!layer.options.mimetype) {
+                // fetch the 1st image
```
Comment (autra) [inline src/Core/Scheduler/Providers/StaticProvider.js]: `// fetch the first image to detect mimetype` ?

**Review by autra — APPROVED** [review]
LGTM, one minor comment if you care to fix :-)

File: src/Core/Scheduler/Providers/StaticProvider.js (hunk 1 above)
Comment (peppsac) [reply src/Core/Scheduler/Providers/StaticProvider.js]: Fixed.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**